### PR TITLE
Feature/xray poc - add an xray decorator to the pykiso tests to import the JUnit results of the marked test to xray (jira plugin)

### PR DESCRIPTION
**Pull request by DianeMornas**
In order to import JUnit test results to xray, create an xray decorator logic to mark pykiso tests so that the results of marked tests can be imported into xray.

### CODE REVIEW

File: docs/whats_new/version_ongoing.rst [hunk 1]
```diff
@@ -1,2 +1,7 @@
 Version ongoing
 ---------------
+
+Results can be exported on Xrax
```
Comment (yannpoupon) [inline docs/whats_new/version_ongoing.rst:4]:
```suggestion
Results can be exported on Xray
```

File: docs/whats_new/version_ongoing.rst [hunk 2]
```diff
@@ -1,2 +1,7 @@
...
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```
Comment (yannpoupon) [inline docs/whats_new/version_ongoing.rst:5]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: examples/test_suite_1/test_suite_1.py [hunk 3]
```diff
@@ -55,6 +55,11 @@ class SuiteTearDown(pykiso.RemoteTestSuiteTeardown):
     yaml configuration file)
     """
 
+    def tearDown(self):
```
Comment (yannpoupon) [inline examples/test_suite_1/test_suite_1.py]: what does this tearDown add to the test, you are checking if True is True.

File: examples/test_suite_1/test_suite_1.py [hunk 4]
```diff
@@ -115,6 +132,7 @@ def tearDown(self):
         we will run the default tearDown()
         """
         super().tearDown()
+        # self.properties = {"test_key": "BDU3-13305"}
```
Comment (yannpoupon) [inline examples/test_suite_1/test_suite_1.py]:
```suggestion
```

File: examples/test_suite_1/test_suite_1.py (hunk 3 above)
Comment (DianeMornas) [reply examples/test_suite_1/test_suite_1.py]: I forgot to remove it ;)

File: docs/tools/xray.rst [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+.. _xray:
+
+Export results on Xray
+======================
+
+The ``xray`` CLI utility takes your Pykiso Junit report and export them on `Xray <https://xray.cloud.getxray.app/>`__.
+
+Upload your results
+-------------------
+To upload your results on Xray users have to follow the command :
```
Comment (Pog3k) [inline docs/tools/xray.rst:11]:
```suggestion
The ``xray`` CLI utility takes your Pykiso Junit report and export them to `Xray <https://xray.cloud.getxray.app/>`__.

Upload your results
-------------------
To upload your results on Xray users have to follow the command :
```

File: docs/whats_new/version_ongoing.rst [hunk 6]
```diff
@@ -14,3 +14,8 @@ PCAN Connector
 By default the CCPCanCan will use the trace size define in during the initialisation.
 
 The log path is now initialise if set at None.
+
+Results can be exported on Xray
```
Comment (Pog3k) [inline docs/whats_new/version_ongoing.rst:18]:
```suggestion
Results can be exported to Xray
```

File: src/pykiso/test_coordinator/test_case.py [hunk 7]
```diff
@@ -317,3 +337,27 @@ def func_wrapper(self: BasicTest) -> None:
         return func_wrapper
 
     return decorator
+
+
+def xray(test_key: str, req_id: str | None = None) -> None:
+    """Decorator: to mark the test to import the JUnit xml results into xray
+
+    :param test_key: the xray ticket id linked to be linked to the test
+    :param requirement: the requirement ticket id to be linked to the test
```
Comment (Pog3k) [inline src/pykiso/test_coordinator/test_case.py:346]:
```suggestion
    :param req_id: the requirement ticket id to be linked to the test
```

File: src/pykiso/tool/xray/cli.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+@click.group()
+@click.option(
+    "--user",
+    help="Xray user id",
+    required=True,
+    default=None,
+    hide_input=True,
+)
+@click.option(
```
Comment (yannpoupon) [inline src/pykiso/tool/xray/cli.py:16]: You could add shortcut so user don't need to type `--password` everytime but could type `-p` for example, same for every option.

**Review by yannpoupon — APPROVED** [review]

**Review by Pog3k — APPROVED** [review]

**Review by yannpoupon — APPROVED** [review]

**Pull request merged**